app/models/commit.rb

```diff
@@ -83,6 +83,19 @@ def refresh_statuses!
     end
   end
 
+  def add_status(status_attributes)
+    previous_status = significant_status
+    statuses.create!(status_attributes)
+    reload
+    new_status = significant_status
+    if (previous_status != new_status) # we only care if something changed
```

Comment:
It may not be the same failure. For instance the previous failure may have been for context foo and the new one is for context bar. We've already sent foo when it came in and now we should send bar. Either (or both) of them may be actionable by the user so they should be informed about them.

I'm not sure what you means by "send the `previous_status` altogether".